sky/jobs/server/core.py

```diff
@@ -476,85 +523,77 @@ def _submit_one(
+                            if isinstance(result, tuple):
+                                return all_job_ids, result[1]
+                            else:
+                                return all_job_ids, None
+                        else:
+                            return result
+                    else:
+                        local_handle = backend_utils.is_controller_accessible(
+                            controller=controller, stopped_message='')
+                        backend = backend_utils.get_backend_from_handle(
+                            local_handle)
+                        assert isinstance(backend, backends.CloudVmRayBackend)
+                        with sky_logging.silent():
+                            backend.sync_file_mounts(
+                                handle=local_handle,
+                                all_file_mounts=controller_task.file_mounts,
+                                storage_mounts=controller_task.storage_mounts)
+                        run_script = controller_task.run
+                        assert isinstance(run_script, str)
+                        # Manually add the env variables to the run script.
+                        # Originally this is done in ray jobs submission but now
+                        # we have to do it manually because there is no ray
+                        # runtime on the API server.
+                        env_cmds = [
+                            f'export {k}={v!r}'
+                            for k, v in controller_task.envs.items()
+                        ]
+                        run_script = '\n'.join(env_cmds + [run_script])
+                        # Dump script for high availability recovery.
+                        for job_id in consolidation_mode_job_ids:
+                            managed_job_state.set_ha_recovery_script(
+                                job_id, run_script)
+                        backend.run_on_head(local_handle, run_script)
+                        ux_utils.starting_message(
+                            f'Job submitted, ID: {consolidation_mode_job_ids}')
+                        return consolidation_mode_job_ids, local_handle
 
     if pool is None:
         if consolidation_mode_job_ids is None:
             return _submit_one()
         assert len(consolidation_mode_job_ids) == 1
-        return _submit_one(consolidation_mode_job_ids[0])
+        return _submit_one(consolidation_mode_job_ids)
 
     ids: List[int] = []
     all_handle: Optional[backends.ResourceHandle] = None
 
-    if num_jobs == 1:
-        job_id = (consolidation_mode_job_ids[0]
-                  if consolidation_mode_job_ids is not None else None)
-        jid, handle = _submit_one(job_id, 0, num_jobs=num_jobs)
-        assert jid is not None, (job_id, handle)
-        ids.append(jid)
-        all_handle = handle
+    # If we have pre_created_job_ids, use single controller task
```

Comment:
We could still here have `pool is not None` but `num_jobs == 1`, in which case we won't have `pre_created_job_ids`, right? May just need to update the comment.